win32ss/user/user32/windows/cursoricon.c

```diff
@@ -1435,21 +1669,16 @@ CURSORICON_LoadFromFileW(
     if(!NtUserSetCursorIconData(hCurIcon, NULL, NULL, &cursorData))
     {
         NtUserDestroyCursor(hCurIcon, TRUE);
-        goto end_error;
+        hCurIcon = NULL;
+        if (cursorData.hbmMask) DeleteObject(cursorData.hbmMask);
```

Comment:
I'm not sure about these changes.

If the very first `CURSORICON_GetCursorDataFromBMI` call succeeded (see the whole new code block added above), then the icon is a regular one (not PNG/Vista), and `cursorData` is initialized to something.
Now, if the `NtUserxCreateEmptyCurObject` call done above fails, we goto end. And in this case, shouldn't we call DeleteObject on these hbm fields, as the original code did?